### PR TITLE
Matchers 2873 v5 Separate Out Matchers

### PR DESCRIPTION
**Pull request by himanshi18037**
Since all the code for matchers happens to be in main.py thus making it quite cluttered therefore a new file matchers.py has all the code for matchers for which has appropriate function calls in main.py.

Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [x] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link
to
https://redmine.openinfosecfoundation.org/issues/2873

Describe changes:
-A new file matchers.py has all the code for matchers which has appropriate function calls in main.py thus, helping in creating separation in logic for different services. Hence, main.py has also been decluttered.

### CODE REVIEW

**Comment by himanshi18037** [comment]
Hey ! @shivan1b 
I found my mistake and this time I have tried to create it correctly please have a look at it because we have less time left to update the link for contribution.

File: suricata/update/main.py [hunk 1]
```diff
@@ -626,7 +416,7 @@ def write_to_directory(directory, files, rulemap):
                         modified.append(rule.id)
         for key in rulemap:
             if not key in oldset:
-                added.append(key)
+                added.append(rule.id)
```
Comment (shivan1b) [inline suricata/update/main.py]: This should not be a part of your commits.

File: suricata/update/main.py [hunk 2]
```diff
@@ -1102,7 +892,7 @@ def _main():
                                help="Filename of drop rules filters")
     
     update_parser.add_argument("--ignore", metavar="<pattern>", action="append",
-                               default=None,
+                               default=[],
```
Comment (shivan1b) [inline suricata/update/main.py]: This is wrong too. You overwrote a commit that is merged into master.

**Comment by shivan1b** [comment]
Replaced by #137

**Pull request closed without merging**